modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm

```diff
@@ -657,9 +698,13 @@
 /obj/item/organ/internal/brain/on_life(delta_time, times_fired) //All your horny is here *points to the head*
 	. = ..()
 	var/mob/living/carbon/human/brain_owner = owner
-	if(istype(brain_owner, /mob/living/carbon/human) && brain_owner.client?.prefs?.read_preference(/datum/preference/toggle/erp/sex_toy))
-		if(!(organ_flags & ORGAN_FAILING))
-			brain_owner.dna.species.handle_arousal(brain_owner, delta_time, times_fired)
+
+	if(!istype(brain_owner, /mob/living/carbon/human) || !(brain_owner.client?.prefs?.read_preference(/datum/preference/toggle/erp/sex_toy)))
+		return FALSE
+	if(!(organ_flags & ORGAN_FAILING))
+		return FALSE
+
+	brain_owner.dna.species.handle_arousal(brain_owner, delta_time, times_fired)
```

Comment:
does this really need to be species level

Reply:
+1 on this, I really don't think it needs to be handled at the species level either.